### PR TITLE
Fix potential Null pointer dereference in test-fdleak.c

### PR DESCRIPTION
**Pull request by icy17**

### CODE REVIEW

File: test/test-fdleak.c [hunk 1]
```diff
@@ -111,6 +111,9 @@ listener_accept_cb(struct evconnlistener *listener, evutil_socket_t sock,
 	struct event_base *base = evconnlistener_get_base(listener);
 	struct bufferevent *bev = bufferevent_socket_new(base, sock,
 		BEV_OPT_CLOSE_ON_FREE);
+	if (bev == NULL) {
+		return;
```
Comment (azat) [inline test/test-fdleak.c:115]: fd leaked here

File: test/test-fdleak.c [hunk 2]
```diff
@@ -209,6 +212,10 @@ start_client(struct event_base *base)
 {
 	struct bufferevent *bev = bufferevent_socket_new(base, -1,
                                                          BEV_OPT_CLOSE_ON_FREE);
+	if (bev == NULL) {
```
Comment (azat) [inline test/test-fdleak.c:215]: I wish that the `main` would have a normal error path, i.e.:

```
if (!start_client)
    goto cleanup;
...
cleanup:
    if (listener)
        evconnlistener_free(listener);
    if (base)
        event_base_free(base);
```

File: test/test-fdleak.c (hunk 1 above)
Comment (icy17) [reply test/test-fdleak.c:115]: OK, I'll fix it

File: test/test-fdleak.c (hunk 2 above)
Comment (icy17) [reply test/test-fdleak.c:215]: OK, thanks for the example

**Comment by icy17** [comment]
I previously misunderstood; while modifying the code, I noticed that I hadn't altered the main function, so perhaps adding a cleanup path is unnecessary? Also, regarding the 'fd leaked' issue, there seems to be no file descriptor here. I believe there are no newly allocated variables within this function, thus no need for cleanup. If I've overlooked something, please highlight it for me.

**Comment by azat** [comment]
>I noticed that I hadn't altered the main function, so perhaps adding a cleanup path is unnecessary

You didn't, but you modified start_client, and main calls it (start_loop -> start_client), so if it can fail it should propagate error to main, since the whole point of this PR to make avoid any leaks on exit AFAIU

>Also, regarding the 'fd leaked' issue, there seems to be no file descriptor here

`listener_accept_cb` accepts socket, if the bufferevent fails you need to close the socket

**Comment by icy17** [comment]
> You didn't, but you modified start_client, and main calls it (start_loop -> start_client), so if it can fail it should propagate error to main, since the whole point of this PR to make avoid any leaks on exit AFAIU

Perhaps I should handle the cleanup in `start_loop` not in `main`? Also, I’m not sure whether to change the exit to return in `start_client`. By doing so, resources could be properly released if `start_client` fails.

**Comment by azat** [comment]
>Perhaps I should handle the cleanup in start_loop not in main?

This is OK

> Also, I’m not sure whether to change the exit to return in start_client. By doing so, resources could be properly released if start_client fails.

Yes, you should not call `exit` from `start_client`

**Comment by azat** [comment]
Please resubmit

**Pull request closed without merging**